File: backend/schema/statistiques.py
from pydantic import BaseModel


class GammeParMoisOut(BaseModel):
    mois: str  # format "AAAA-MM"
    nb_gammes: int


class ArticleParChaineOut(BaseModel):
    chaine: str
    nb_articles: int


class StatistiquesOut(BaseModel):
    nb_articles: int
    nb_gammes: int
    nb_operateurs_actifs: int
    nb_chaines: int
    smv_moyen_minutes: float | None = None
    gammes_par_mois: list[GammeParMoisOut]
    articles_par_chaine: list[ArticleParChaineOut]

File: frontend/src/types/statistiques.ts
export interface GammeParMois {
  mois: string
  nb_gammes: number
}

export interface ArticleParChaine {
  chaine: string
  nb_articles: number
}

export interface Statistiques {
  nb_articles: number
  nb_gammes: number
  nb_operateurs_actifs: number
  nb_chaines: number
  smv_moyen_minutes: number | null
  gammes_par_mois: GammeParMois[]
  articles_par_chaine: ArticleParChaine[]
}
